engine/src/main/java/org/destinationsol/body/components/BodyLinked.java

```diff
@@ -16,10 +16,37 @@
 package org.destinationsol.body.components;
 
 import com.badlogic.gdx.physics.box2d.Body;
-import org.terasology.gestalt.entitysystem.component.EmptyComponent;
+import org.destinationsol.body.systems.BodyHandlerSystem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.gestalt.entitysystem.component.Component;
+
 
 /**
- * Indicates that there is a {@link Body} associated with the entity.
+ * Indicates that there is a {@link Body} associated with the entity. It also contains the mass of the entity.
  */
-public class BodyLinked extends EmptyComponent<BodyLinked> {
+public class BodyLinked implements Component<BodyLinked> {
+    private static final Logger logger = LoggerFactory.getLogger(BodyLinked.class);
+
+    private float mass;
+
+    /**
+     * Sets the mass of the entity. This is called every tick by the {@link BodyHandlerSystem}.
+     */
+    public void setMass(float mass) {
+        if (mass <= 0) {
+            logger.error("Invalid value for the mass. It can't be less than or equal to zero.");
+            this.mass = 1;
+        }
+        this.mass = mass;
+    }
```

Comment:
You set `this.mass` to 1 if the parameter is a bad value and then proceed to overwrite it with the bad value :wink: